cohere/client.py

```diff
@@ -1002,12 +1008,26 @@ def create_custom_model(
 
         """
         internal_custom_model_type = CUSTOM_MODEL_PRODUCT_MAPPING[model_type]
+
+        # Figuring out base model
+        if internal_custom_model_type in ["GENERATIVE", "CLASSIFICATION"]:
+            assert base_model is None, "base_model has to be None for generative and classification models"
```

Comment:
this validation should happen on the backend

Reply:
Oh we had a discussion with Inna and Raphael, and we thought that it made more sense to do an internal mapping here like it's done for `internal_custom_model_type` with `internal_custom_model_type = CUSTOM_MODEL_PRODUCT_MAPPING[model_type]`

The other solution without the mapping would be to:
- use `base_model: str = "medium"` as default argument of the function
- this would force the user to specify `base_model="english"` or `base_model="multilingual"` for reranker.

What do you think is the best solution?

Reply:
- for non-rerank finetuning, we want to continue supporting `None` as a value for the base_model. That allows us to change the default value on the backend.

- for rerank finetuning, the backend should throw an error "base_model cannot be empty for finetune type "rerank". Must be one of "english" or "multilingual"

no need to keep a map of defaults on the client

Reply:
For non-rerank finetuning, the base_model value has always been "medium": https://github.com/cohere-ai/cohere-python/blob/d21c04bab8e2de3960012e928c3ff7dcd42b9869/cohere/client.py#L1120
-> that's why I'm thinking of setting "medium" as the default value (ensures no breaking change) if we don't keep a map of defaults on the client

Reply:
just like for all other apis, we should not be setting any defaults in the client, so just remove that "medium" setting from the SDK and make sure that the backend sets the correct value

Reply:
@raphaelcristal tagging you here as this is probably something you own. Do you think it's possible to do what @mkozakov is suggesting above? I.e. remove the `"baseModel":"medium"` default and instead add defaults to the backend?

Reply:
Yes what Michael says is correct. Best to keep all validation on the backend, so we don't have to deal with deprecations in the sdk. Sorry if I didn't explain this properly when we talked.